packages/block-library/src/site-logo/edit.js

```diff
@@ -416,10 +416,12 @@ export default function LogoEdit( {
 		} );
 	};
 
-	const setIcon = ( newValue ) =>
+	const setIcon = ( newValue ) => {
+		newValue = newValue === undefined ? null : newValue;
```

Comment:
```suggestion
		// The new value needs to be `null` to reset the Site Icon.
```

Let's leave comments here to avoid regressions.

Reply:
Thank you for you suggestion. ill include this.